### PR TITLE
Fixes #10703 - websockify is now randomly selected

### PR DESCRIPTION
**Pull request by lzap**
Plus updated websockify to 1.6.1 version

### CODE REVIEW

**Comment by dLobatog** [comment]
@lzap minor errors in Rubocop

File: lib/ws_proxy.rb [hunk 1]
```diff
@@ -21,22 +23,36 @@ def self.start(attributes)
...
-      # if the port is already in use, try another one from the pool
-      # this is not ideal, as it would try all ports in order
-      # but it avoids any threading issues etc.
-      # TODO: try to select a port from a pool randomly, so we always hit all active connections.
     rescue PortInUse
+      # fallback just in case of race condition
       port += 1
-      retry if port <= PORTS.last
+      if port >= PORTS.last
+        raise ::Foreman::Exception, 'No free ports for websockify, try later'
```
Comment (domcleal) [inline lib/ws_proxy.rb]: extract

**Comment by lzap** [comment]
All fixed, let's rock!

**Comment by dLobatog** [comment]
Merged as 67f9b68e2d2cc1931b08cb7dd091a5343bfe4904, thanks @lzap!

**Pull request closed without merging**